apps/frontend/src/hooks/common/useDataApi.ts

```diff
@@ -15,6 +15,26 @@ import { RequestQuery } from 'utils/utilTypes';
 
 const endpoint = '/graphql';
 
+const clientNameHeader = 'apollographql-client-name';
+const anonymousClientName = 'UOP frontend - anonymous';
+function setAnonymousClientNameHeader(client: GraphQLClient) {
+  return client.setHeader(clientNameHeader, anonymousClientName);
+}
+function getClientNameHeaderSetter(
+  includeUserId: boolean,
+  userId?: number | undefined
+) {
+  if (includeUserId) {
+    return (client: GraphQLClient) =>
+      client.setHeader(
+        clientNameHeader,
+        userId ? `UOP frontend - user ${userId}` : anonymousClientName
+      );
+  } else {
+    return setAnonymousClientNameHeader;

```

Comment:
I would not add anything to the header if the feature is turned off. Could we use this?:
```suggestion
    return (client) => client;
```

Reply:
Sorry, I thought the feature you were after was wether to include user numbers when available, and if the flag's not set always use the anonymous header.

I think that'd put us in a similar situation to setting the header on the backend. When determining whether to set the header based on a feature flag we need to make at least that one API request to get those feature flags. Not setting the anonymous header by default would mean that there's still unidentified requests being made that can't be attributed to the frontend in the analytics.

Reply:
@janosbabik I'd really like to have all our requests identified in the analytics where possible. If the default header was just "UOP frontend", instead of "UOP frontend - anonymous" would that be acceptable?

Reply:
@ACLay Yes, I can work with this, but I have one last thought. I'm not sticking to the backend (BE) solution, but there are a few things to keep in mind:
With this approach, the server can receive any header values without validation, meaning I could send requests on behalf of other users or add custom messages. It’s not a huge issue, but it doesn’t feel thrustworthy.
If we handle this on the backend, you’d be able to control and validate incoming requests using user tokens and API tokens, which would cover all cases. Also with DI it can be turned off. 
I know that modifying request headers can seem hacky, but the frontend (FE) solution is a bit unconventional too.

Reply:
How about I always set an anonymous "UOP frontend" header value in the frontend, then in the backend, if the header value starts with that I append user data from the token when available? That way we identify all calls, without being entirely trusting of data that a users browser could manipulate